### PR TITLE
Magic Link Enhancements

### PR DESCRIPTION
**Pull request by jrmccannon**
- Adding TTL to magic link request
- Changing DTO to allow for TTL time span.
- Added MagicLinkToken method with `magic_link` type

### CODE REVIEW

File: src/Service/MagicLinks/Models/SendMagicLinkRequest.cs [hunk 1]
```diff
@@ -17,10 +18,10 @@ public class SendMagicLinkRequest
...
-        UserId = UserId,
-        EmailAddress = new MailAddress(EmailAddress),
-        LinkTemplate = UrlTemplate
-    };
+    /// <summary>
+    /// Number of seconds the magic link will be valid for.
+    /// </summary>
+    public int? TimeToLive { get; init; }
+
+    public MagicLinkTokenRequest ToDto() => new(UserId, new MailAddress(EmailAddress), UrlTemplate);
```
Comment (abergs) [inline src/Service/MagicLinks/Models/SendMagicLinkRequest.cs:26]: maybe issue: You are not using the TimeToLive field.

File: src/Service/Models/MagicLinkTokenRequest.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+/// </summary>
+/// <param name="userId">User identifier for the recipient of the magic link.</param>
+/// <param name="emailAddress">Email address for the intended user.</param>
+/// <param name="linkTemplate">Template used for creating the magic link. Token template string will be swapped with actual token.</param>
+/// <param name="timeToLive">Time span the magic link will be valid for. Default lifespan is 15 minutes.</param>
+public class MagicLinkTokenRequest(string userId, MailAddress emailAddress, string linkTemplate, TimeSpan timeToLive) : RequestBase
+{
+    public MagicLinkTokenRequest(string userId, MailAddress emailAddress, string linkTemplate)
+        : this(userId, emailAddress, linkTemplate, new TimeSpan(0, 15, 0)) { }
+    public MagicLinkTokenRequest(string userId, MailAddress emailAddress, string linkTemplate, int timeToLive)
```
Comment (abergs) [inline src/Service/Models/MagicLinkTokenRequest.cs:16]: this constructor is currently never used

**Review by abergs — APPROVED** [review]
See comments, but if they are non-issue this is approved

File: src/Service/MagicLinks/Models/SendMagicLinkRequest.cs (hunk 1 above)
Comment (jrmccannon) [reply src/Service/MagicLinks/Models/SendMagicLinkRequest.cs]: Fixed this.

File: src/Service/Models/MagicLinkTokenRequest.cs (hunk 2 above)
Comment (jrmccannon) [reply src/Service/Models/MagicLinkTokenRequest.cs:16]: Fixed this

**Pull request merged**